### PR TITLE
Fix for math-related deprecations

### PR DESCRIPTION
**Pull request by Br1ght0ne**
Because of https://github.com/CleverRaven/Cataclysm-DDA/pull/71286, `arithmetic` and `adjust_var` used by this mod no longer work. I attempted to port these to `math` syntax as best as I could.
This is my first encounter with EOCs, so this PR might require revisions.

Incorporates `time` fixes from https://github.com/TGWeaver/CDDA-Sky-Islands/pull/57 (thanks @bcmcms!)

### CODE REVIEW

**Comment by Br1ght0ne** [comment]
My bad: I did not see #55 and #57 before opening this PR. I included the patch by @bcmcms in this PR, but if any of the previous PRs get merged, I'll update my PR.

**Comment by Br1ght0ne** [comment]
Needs to be fixed:

<details>
<summary>errors related to `u_val('missionswon')`</summary>

```
 DEBUG    : error setting double destination in 'u_val: missionswon', 

 FUNCTION : static void kwargs_shim::throw_error(std::string_view)
 FILE     : src/math_parser_shim.h
 LINE     : 53
 VERSION  : 1949d26
```

```
 DEBUG    : unrecognized number source in 'u_val: missionswon', 

 FUNCTION : static void kwargs_shim::throw_error(std::string_view)
 FILE     : src/math_parser_shim.h
 LINE     : 53
 VERSION  : 1949d26
```
</details>

**Comment by DenCheesecake** [comment]
@Br1ght0ne thank you for the update! How critical are the errors related to `u_val('missionswon')`? Is the current build still playable with these?

File: missions_and_mapgen.json [hunk 1]
```diff
@@ -976,15 +902,9 @@
     },
     "end": {
       "effect": [
-        { "u_adjust_var": "missionswon", "adjustment": 1 },
+        { "math": [ "u_val('missionswon')", "++" ] },
```
Comment (aucun6352) [inline missions_and_mapgen.json:905]:
```suggestion
        { "math": [ "u_missionswon", "++" ] },

```

File: missions_and_mapgen.json [hunk 2]
```diff
@@ -1077,7 +983,7 @@
     "monster_kill_goal": 10,
     "difficulty": 0,
     "value": 0,
-    "end": { "effect": [ { "u_adjust_var": "slaughterswon", "adjustment": 1 }, { "u_spawn_item": "warptoken", "count": 1 } ] }
+    "end": { "effect": [ { "math": [ "u_val('slaughterswon')", "++" ] }, { "u_spawn_item": "warptoken", "count": 1 } ] }
```
Comment (aucun6352) [inline missions_and_mapgen.json:986]:
```suggestion
    "end": { "effect": [ { "math": [ "slaughterswon", "++" ] }, { "u_spawn_item": "warptoken", "count": 1 } ] }

```

Comment (aucun6352) [reply missions_and_mapgen.json:986]: [Cataclysm-DDA#71027](https://github.com/CleverRaven/Cataclysm-DDA/pull/71027/files#diff-d0d00e2f86303f379e42d9c96a4a84f29e6e67e17262388a9f4c639f023c1b42R916)

File: missions_and_mapgen.json (hunk 1 above)
Comment (aucun6352) [reply missions_and_mapgen.json:905]: [Cataclysm-DDA#71027](https://github.com/CleverRaven/Cataclysm-DDA/pull/71027/files#diff-d0d00e2f86303f379e42d9c96a4a84f29e6e67e17262388a9f4c639f023c1b42R896)

**Comment by Br1ght0ne** [comment]
Thank you @aucun6352!

Re @DenCheesecake: the build was playable, though the "missions won" number in expedition stats was broken. Now, thanks to @aucun6352, everything should work and throw no errors!

**Comment by Br1ght0ne** [comment]
Handled by https://github.com/CleverRaven/Cataclysm-DDA/pull/71496.

**Pull request closed without merging**